addons/survey/models/survey_question.py

```diff
@@ -274,29 +277,162 @@ def _index(self):
+
+    def _get_stats_data_answers(self, user_input_lines):
+        """ [{"text": "Once a day", "count": 0, "answer_id": 1, "answer_score": 0.0}, ...] """
+        suggested_answers = [answer for answer in self.mapped('suggested_answer_ids')]
+        if self.comment_count_as_answer:
+            suggested_answers += [self.env['survey.question.answer']]
+
+        count_data = dict.fromkeys(suggested_answers, 0)
+        for line in user_input_lines:
+            if line.suggested_answer_id or (line.value_char_box and self.comment_count_as_answer):
+                count_data[line.suggested_answer_id] += 1
+
+        table_data = [{
+            'value': _('Other (see comments)') if not sug_answer else sug_answer.value,
+            'suggested_answer': sug_answer,
+            'count': count_data[sug_answer]
+            }
+            for sug_answer in suggested_answers]
+        graph_data = [{
+            'text': _('Other (see comments)') if not sug_answer else sug_answer.value,
+            'count': count_data[sug_answer]
+            }
+            for sug_answer in suggested_answers]
+
+        return table_data, graph_data
+
+    def _get_stats_graph_data_matrix(self, user_input_lines):
+        """ [{"key": "which blah", "values": [{"text": "High quality", "count": 1, "answer_id": 5, "answer_score": 0.0}, ... }] """
+        suggested_answers = self.mapped('suggested_answer_ids')
+        matrix_rows = self.mapped('matrix_row_ids')
+
+        count_data = dict.fromkeys(itertools.product(matrix_rows, suggested_answers), 0)
+        for line in user_input_lines:
+            if line.matrix_row_id and line.suggested_answer_id:
+                count_data[(line.matrix_row_id, line.suggested_answer_id)] += 1
+
+        table_data = [{
+            'row': row,
+            'columns': [{
+                'suggested_answer': sug_answer,
+                'count': count_data[(row, sug_answer)]
+            } for sug_answer in suggested_answers],
+        } for row in matrix_rows]
+        graph_data = [{
+            'key': sug_answer.value,
+            'values': [{
+                'text': row.value,
+                'count': count_data[(row, sug_answer)]
+                }
+                for row in matrix_rows
+            ]
+        } for sug_answer in suggested_answers]
+
+        return table_data, graph_data
+
+    def _get_stats_summary_data(self, user_input_lines):
+        if self.question_type in ['simple_choice', 'multiple_choice']:
+            return self._get_stats_summary_data_choice(user_input_lines)
+        if self.question_type in ['numerical_box']:
+            return self._get_stats_summary_data_numerical(user_input_lines)
+        return {}
+
+    def _get_stats_summary_data_choice(self, user_input_lines):
+        right_answers = self.suggested_answer_ids.filtered(lambda label: label.is_correct)
+        right_inputs = user_input_lines.filtered(lambda line: line.answer_is_correct).mapped('user_input_id')
```

Comment:
For a multiple choice, I would say that you need to have all the right choices (and only the right choices) to be considered as a "right_input"

Reply:
I wondered if we should not have "right, partial, failed" . I will give it a try.